eng/pipelines/OpenApiSpecs/repo-structure-cache-job.yml

```diff
@@ -0,0 +1,39 @@
+# Starter pipeline
```

Comment:
Lets move this directly into the eng/pipelines directory and name change the name slightly. I'm assuming this is only about caching the spec information so perhaps `cache-specs-in-spec-repo.yml` might be slightly better name to represent what it is happening.